base/applications/rapps/include/settings.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+#include <windef.h>
+#include <wininet.h>
+
+struct SETTINGS_INFO
+{
+    BOOL bSaveWndPos;
+    BOOL bUpdateAtStart;
+    BOOL bLogEnabled;
+    WCHAR szDownloadDir[MAX_PATH];
+    BOOL bDelInstaller;
+    /* Window Pos */
+    BOOL Maximized;
+    INT Left;
+    INT Top;
+    INT Width;
+    INT Height;
+    /* Proxy settings */
+    INT Proxy;
+    WCHAR szProxyServer[MAX_PATH];
+    WCHAR szNoProxyFor[MAX_PATH];
+    /* Software source settings */
+    BOOL bUseSource;
+    WCHAR szSourceURL[INTERNET_MAX_URL_LENGTH];
+};
+
+typedef SETTINGS_INFO *PSETTINGS_INFO;
+
+enum SETTINGS_FIELDTYPE
+{
+    SettingsFieldInt,
+    SettingsFieldBool,
+    SettingsFieldString,
+};
+
+typedef struct _SettingsRegEntry
+{
+    size_t Offset;  // Offset of the field in SETTINGS_INFO struct
```

Comment:
If you change this to a union something like this:
```
union
{
    BOOL* bValue;
    INT* iValue;
    WCHAR* wValue;
};
```
you can create pointers to the settings instead of using the `OFFSET_OF` macro

Reply:
I probably didn't gey your means.... then how should I know where's the field I want to visit? 🤣